table/tables/index.go

```diff
@@ -130,6 +130,26 @@ func (c *index) GenIndexKey(indexedValues []types.Datum, h int64) (key []byte, d
 			}
 		}
 	}
+
+	if len(indexedValues) > 0 {
```

Comment:
I  think we should also modify get row by index related code. Now the index is not exactly match, it is prefix match. So we should check if conditions are matched after we get row data.